src/time.scss

```diff
@@ -7,31 +7,221 @@ $block: #{$fd-namespace}-time;
+    width: 3rem;
+    height: $fd-time-item-height;
+    border: 0.0625rem solid var(--sapList_SelectionBorderColor);
+    border-radius: var(--sapButton_BorderCornerRadius);
+  }
+
+  &__wrapper {
+    @include fd-reset();
+
+    overflow: hidden;
+    position: relative;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    flex: 1 1 auto;
+    width: 3rem;
+
+    &--active {
+      max-height: $fd-time-wrapper-height;
+      height: $fd-time-wrapper-height;
+      justify-content: initial;
+    }
+
+    &--meridian {
+      .#{$block}__item {
+        &:first-child {
+          margin-top: $fd-time-item-height * 3;
+        }
+
+        &:last-child {
+          margin-bottom: $fd-time-item-height * 3;
+        }
+      }
 
-    // Spacing and narrower side padding for child inputs
-    .#{$block}__input {
-      margin: fd-space(2) 0;
-      padding-left: fd-space(2);
-      padding-right: fd-space(2);
-      text-align: center;
-      width: 100%;
+      .#{$block}__list {
+        height: $fd-time-wrapper-height;
+      }
     }
   }
 
+  &__item {
+    @include fd-reset();
+
+    cursor: default;
+    width: 3rem;
+    height: $fd-time-item-height;
+    min-height: $fd-time-item-height;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border-radius: var(--sapButton_BorderCornerRadius);
+    background-color: var(--sapLegend_WorkingBackground);
+    border: 0.0625rem solid var(--sapList_Background);
+
+    &:hover {
+      background-color: var(--sapList_Hover_Background);
+    }
+
+    &:active {
+      background-color: var(--sapContent_Selected_Background);
+      border-color: transparent;
+      color: var(--sapContent_Selected_TextColor);
+    }
+
+    &--current {
+      border-color: var(--sapList_SelectionBorderColor);
+      background-color: var(--sapList_Background);
+
+      &:hover {
+        background-color: var(--sapList_Hover_Background);
+      }
+
+      &:active {
+        background-color: var(--sapContent_Selected_Background);
+        border-color: transparent;
+        color: var(--sapContent_Selected_TextColor);
+      }
+    }
+
+    &--collapsed {
+      background-color: var(--sapContent_Selected_Background);
+      color: var(--sapContent_Selected_TextColor);
+      border-color: transparent;
+
+      &:hover {
+        background-color: var(--sapContent_Selected_Hover_Background);
+      }
+    }
+
+    &--hidden {
+      visibility: hidden;
+    }
+  }
+
+  &__unit {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: 100%;
+    height: 100%;
+    z-index: 2;
+  }
+
+  &__list {
+    @include fd-reset();
+    @include fd-remove-scrollbar();
+
+    overflow: hidden;
+  }
+
+  &__slider-label {
+    @include fd-reset();
+
+    line-height: $fd-time-item-compact-height;
+    height: $fd-time-item-compact-height;
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
```

Comment:
we need to define a max-width for the overflow, ask the designers

Reply:
Hello @stefan-sto I think max width should be `100%` there

Reply:
@JKMarkowski wrong Stefano